src/magic_folder/config.py

```diff
@@ -0,0 +1,419 @@
+"""
+Configuration and state database interaction.
+
+See also docs/config.rst
+"""
+
+from os import (
+    mkdir,
+    urandom,
+)
+from base64 import (
+    urlsafe_b64encode,
+)
+
+import attr
+
+import sqlite3
+
+from nacl.signing import (
+    SigningKey,
+)
+from nacl.encoding import (
+    Base32Encoder,
+)
+
+from twisted.internet.endpoints import (
+    serverFromString,
+)
+from twisted.python.filepath import (
+    FilePath,
+)
+
+from .snapshot import (
+    LocalAuthor,
+)
+
+
+class ConfigurationError(Exception):
+    """
+    The configuration is unusable for some reason
+    """
+
+
+_global_config_version = 1
+
+_global_config_schema = """
+CREATE TABLE version
+(
+    version INTEGER  -- contains one row, set to 1
+);
+
+CREATE TABLE magic_folders
+(
+    name          TEXT PRIMARY KEY,  -- UTF8 name of this folder
+    location      TEXT               -- UTF8 path to this folder's configuration/state
+);
+
+CREATE TABLE config
+(
+    api_endpoint TEXT,                -- Twisted server-string for our HTTP API
+    tahoe_client_url TEXT            -- HTTP URL of our Tahoe-LAFS client
+);
+"""
+
+_magicfolder_config_version = 1
+
+_magicfolder_config_schema = """
+CREATE TABLE version
+(
+    version INTEGER  -- contains one row, set to 1
+);
+
+CREATE TABLE stash
+(
+    path TEXT    -- the path to our stash-directory
+);
```

Comment:
Is this a left-over?  It seels like `config.stash_path` represents the same thing.